pkg/apis/seedmanagement/v1alpha1/defaults_managedseed.go

```diff
@@ -103,6 +104,19 @@ func setDefaultsGardenlet(obj *Gardenlet, name, namespace string) {
 		}
 	}
 
+	// Set gardenlet deployment defaults for high availability.
+	if gardenletConfig.SeedConfig != nil {
+		if _, ok := gardenletConfig.SeedConfig.Labels[v1beta1constants.LabelSeedMultiZonal]; ok {
+			if obj.Deployment.FailureToleranceType == nil {
+				failureToleranceZone := gardencorev1beta1.FailureToleranceTypeZone
+				obj.Deployment.FailureToleranceType = &failureToleranceZone
+			}
+			if obj.Deployment.ReplicaCount == nil {
+				obj.Deployment.ReplicaCount = pointer.Int32(2)
+			}
+		}
```

Comment:
Should we wait for https://github.com/gardener/gardener/pull/6723? If not, can you add a TODO that this new `spec` field needs to be evaluated here as well (and add it to your umbrella issue as new task))?